sdk/v2/internal/restmachinery/client_test.go

```diff
@@ -0,0 +1,290 @@
+			assertions: func(t *testing.T, resp *http.Response, err error) {
+				require.NoError(t, err)
+				require.NotNil(t, resp)
+			},
+		},
+		{
+			name: "with query params",
+			req: OutboundRequest{
+				QueryParams: map[string]string{
+					"marco": "polo",
+				},
+			},
+			respStatus: http.StatusOK,
+			assertions: func(t *testing.T, resp *http.Response, err error) {
+				require.NoError(t, err)
+				require.NotNil(t, resp)
+			},
+		},
+		{
+			name:       "with authn error",
+			req:        OutboundRequest{},
+			respStatus: http.StatusUnauthorized,
+			assertions: func(t *testing.T, resp *http.Response, err error) {
+				require.IsType(t, &meta.ErrAuthentication{}, err)
+			},
+		},
+		{
+			name:       "with authz error",
+			req:        OutboundRequest{},
+			respStatus: http.StatusForbidden,
+			assertions: func(t *testing.T, resp *http.Response, err error) {
+				require.IsType(t, &meta.ErrAuthorization{}, err)
+			},
+		},
+		{
+			name:       "with bad request",
+			req:        OutboundRequest{},
+			respStatus: http.StatusBadRequest,
+			assertions: func(t *testing.T, resp *http.Response, err error) {
+				require.IsType(t, &meta.ErrBadRequest{}, err)
+			},
+		},
+		{
+			name:       "with not found",
+			req:        OutboundRequest{},
+			respStatus: http.StatusNotFound,
+			assertions: func(t *testing.T, resp *http.Response, err error) {
+				require.IsType(t, &meta.ErrNotFound{}, err)
+			},
+		},
+		{
+			name:       "with conflict",
+			req:        OutboundRequest{},
+			respStatus: http.StatusConflict,
+			assertions: func(t *testing.T, resp *http.Response, err error) {
+				require.IsType(t, &meta.ErrConflict{}, err)
+			},
+		},
+		{
+			name:       "with internal server error",
+			req:        OutboundRequest{},
+			respStatus: http.StatusInternalServerError,
+			assertions: func(t *testing.T, resp *http.Response, err error) {
+				require.IsType(t, &meta.ErrInternalServer{}, err)
+			},
+		},
+		{
+			name:       "with other error",
+			req:        OutboundRequest{},
+			respStatus: http.StatusBadGateway,
+			assertions: func(t *testing.T, resp *http.Response, err error) {
+				require.Equal(
+					t,
+					fmt.Sprintf("received %d from API server", http.StatusBadGateway),
+					err.Error(),
+				)
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			server := httptest.NewServer(
+				http.HandlerFunc(
+					func(w http.ResponseWriter, r *http.Request) {
+						w.WriteHeader(testCase.respStatus)
+						fmt.Fprintln(w, "{}")
+					},
+				),
+			)
+			client := BaseClient{
+				APIAddress: server.URL,
+				APIToken:   "11235813213455",
+				HTTPClient: &http.Client{
+					Transport: &http.Transport{
+						TLSClientConfig: &tls.Config{
+							InsecureSkipVerify: true,
+						},
+					},
+				},
+			}
+			resp, err := client.SubmitRequest(context.Background(), testCase.req)
+			testCase.assertions(t, resp, err)
+		})
+	}
+}
+
+// 	testProject := Project{
```

Comment:
Are these commented-out lines still intended to be included?

Reply:
Nope... and already removed in a recent commit, iirc, so I'll mark this as resolved.